dds/DCPS/DiscoveryBase.h

```diff
@@ -253,15 +442,15 @@ namespace OpenDDS {
             discovered_subscriptions_.find(to_ignore);
           if (iter != discovered_subscriptions_.end()) {
             // clean up tracking info
-            topics_[get_topic_name(iter->second)].endpoints_.erase(iter->first);
-            remove_from_bit(iter->second);
             OPENDDS_STRING topic_name = get_topic_name(iter->second);
+            TopicDetails& td = topics_[topic_name];
+            td.remove_pub_sub(iter->first);
+            remove_from_bit(iter->second);
             discovered_subscriptions_.erase(iter);
             // break associations
-            typename OPENDDS_MAP(OPENDDS_STRING, TopicDetails)::iterator top_it =
-              topics_.find(topic_name);
-            if (top_it != topics_.end()) {
-              match_endpoints(to_ignore, top_it->second, true /*remove*/);
+            match_endpoints(to_ignore, td, true /*remove*/);
+            if (td.is_dead()) {
+              purge_dead_topic(topic_name);
             }
             return;
```

Comment:
The extra {}'s here feel unnecessary and, in addition to the extra returns everywhere, make this code seem more confusing than it really is.

Reply:
The style guide says I should use {}s.